src/data/fintype/card.lean

```diff
@@ -116,3 +115,48 @@ begin
     ext a ha,
     exact (congr_fun hfg a : _) }
 end
+
+/-- Summing `a^s.card * b^(n-s.card)` over all finite subsets `s` of a fintype of cardinality `n`
+gives `(a + b)^n`. The "good" proof involves expanding along all coordinates using the fact that
+`x^n` is multilinear, but multilinear maps are only available now over rings, so we give instead
+a proof reducing to the usual binomial theorem to have a result over semirings. -/
+lemma fintype.sum_pow_mul_eq_add_pow
```

Comment:
I did a bit of a refactor of this lemma yesterday. I first proved the lemma `prod_add {α R : Type*} [comm_semiring R] (f g : α → R) (s : finset α) : s.prod (λ a, f a + g a) = s.powerset.sum (λ t : finset α, t.prod f * (s.filter (∉ t)).prod g)` and restated this lemma using a `finset α` instead of the whole of `α` and `finset.powerset` in place of `finset.univ`, and proving it using `prod_add`. Are you happy for me to push these changes to this branch?

Reply:
Sure, thanks!